sdk/core/azure-core/azure/core/pipeline/_base_async.py

```diff
@@ -62,11 +67,12 @@ async def send(self, request: PipelineRequest) -> PipelineResponse:
         :rtype: ~azure.core.pipeline.PipelineResponse
         """
         await _await_result(self._policy.on_request, request)
+        response: PipelineResponse[HTTPRequestType, AsyncHTTPResponseType]
         try:
-            response = await self.next.send(request)  # type: ignore
+            response = await self.next.send(request)
         except Exception:  # pylint: disable=broad-except
-            if not await _await_result(self._policy.on_exception, request):
-                raise
+            await _await_result(self._policy.on_exception, request)
```

Comment:
Doesn't this change the behaviour? Or did `on_exception` never return anything in the first place?

Reply:
I've just noticed that it seems odd that `on_exception` doesn't pass the actual exception...

Reply:
And on_exception only returns None. https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/core/azure-core/azure/core/pipeline/policies/_base.py#L118